cdap-source-control/src/test/java/io/cdap/cdap/sourcecontrol/RepositoryManagerTest.java

```diff
@@ -147,14 +147,208 @@ public void testValidateNullBranchName() throws IOException {
+      .call();
+  }
+
+  /**
+   * Adds a file to the git repository hosted by the {@link LocalGitServer}.
+   *
+   * @param relativePath path relative to git repo root.
+   * @param contents     the contents of the file.
+   */
+  private void addFileToGit(Path relativePath, String contents) throws GitAPIException, IOException {
+    Path tempDirPath = tempFolder.newFolder("temp-local-git").toPath();
+    Git localGit = getClonedGit(tempDirPath);
+    Path absolutePath = tempDirPath.resolve(relativePath);
+    // Create parent directories if they don't exist.
+    Files.createDirectories(absolutePath.getParent());
+    Files.write(absolutePath, contents.getBytes(StandardCharsets.UTF_8));
+    localGit.add().addFilepattern(".").call();
+    localGit.commit().setMessage("Adding " + relativePath).call();
+    localGit.push()
+      .setTimeout(GIT_COMMAND_TIMEOUT)
+      .setCredentialsProvider(new UsernamePasswordCredentialsProvider(GIT_SERVER_USERNAME, MOCK_TOKEN))
+      .call();
+    localGit.close();
+    DirUtils.deleteDirectoryContents(tempDirPath.toFile());
+  }
+
+  /**
+   * Adds a symbolic link to the git repository hosted by the {@link LocalGitServer}.
+   *
+   * @param relativeLinkPath   relative path to create the symlink.
+   * @param relativeTargetPath relative path to an existing file in the Git repository.
+   */
+  private void addSymbolicLinkToGit(Path relativeLinkPath, Path relativeTargetPath) throws GitAPIException,
+    IOException {
+    Path tempDirPath = tempFolder.newFolder("temp-local-git").toPath();
+    Git localGit = getClonedGit(tempDirPath);
+    Path absoluteTargetPath = tempDirPath.resolve(relativeTargetPath);
+    Path absoluteLinkPath = tempDirPath.resolve(relativeLinkPath);
+    // Create parent directories for the link if they don't exist.
+    Files.createDirectories(absoluteLinkPath.getParent());
+    // Create a relative link so that it remains valid irrespective of where the repository is cloned.
+    Files.createSymbolicLink(absoluteLinkPath, absoluteLinkPath.getParent().relativize(absoluteTargetPath));
+    localGit.add().addFilepattern(".").call();
+    localGit.commit().setMessage("Adding Symbolic link: " + relativeTargetPath).call();
+    localGit.push()
+      .setTimeout(GIT_COMMAND_TIMEOUT)
+      .setCredentialsProvider(new UsernamePasswordCredentialsProvider(GIT_SERVER_USERNAME, MOCK_TOKEN))
+      .call();
+    localGit.close();
+    DirUtils.deleteDirectoryContents(tempDirPath.toFile());
+  }
+
+  /**
+   * Calculates the hash for provided file contents in a similar way to Git.
+   */
+  private String getGitStyleHash(String fileContents) {
+    // Git prefixes the object with "blob ", followed by the length (as a human-readable integer), followed by a NUL
+    // character and takes the sha1 hash to find the file hash.
+    // See https://stackoverflow.com/a/7225329.
+    return Hashing.sha1().hashString("blob 16\0" + fileContents, StandardCharsets.UTF_8).toString();
```

Comment:
nit: get length from fileContents, not fixed

Reply:
Done.